### ISSUE TITLE
block-no-empty rule

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint-config-standard#1]
I would expect `block-no-empty: true` to be in there. @jeddy3, Did you consider that rule controversial in some way?

**Comment by jeddy3** [comment]
> Did you consider that rule controversial in some way?

Yep, I did. I've seen a couple of BEM/SuitCSS projects that use empty component classes to bring a consistent structure to each file i.e. even if the main component class doesn't have any styles, it still gets added to the top of the file for consistency e.g:

```css
.Component {}

.Component-part {
  top: 0;
}
```

All the rules but two—`rule-no-shorthand-property-overrides` & `color-no-invalid-hex`—in the config are stylistic (and based on the common rules from those style guides I list in the README). I also included the two aforementioned "subtle gotcha" rules as I can't think of a single instance when you'd want an invalid hex or to clobber a previous declaration with a shorthand, whereas I know of at least one use for empty blocks. And so I didn't include it.

It seemed like a good place to draw a line in the sand about what type of rules made the config and what didn't. I think it's probably a good idea that we have a criteria that we can communicate as I'm sure there'll be more queries about rules in this config in the future. What do you think, is it a good enough criteria or shall we revise it?

**Comment by davidtheclark** [comment]
> It seemed like a good place to draw a line in the sand

Could you clarify: what is the line? What is the principle that differentiates?

>  do we instead build a config that we think we be the most useful to the majority of users, even if that infringes on an edge-case

Isn't that what this config is doing, ineluctably?

**Comment by jeddy3** [comment]
> Isn't that what this config is doing, ineluctably?

Good point. I had to look up the word "ineluctably" btw... :)

> Could you clarify: what is the line? What is the principle that differentiates?

It's a bit of a rubbish line when I think about it some more; a lot of waffle in that last message. Lets include the `block-no-empty` rule as it'll definitely help the majority of users.

**Comment by davidtheclark** [comment]
Ha ... well I agree with the intention here, completely: the goal should be to create a config that most users will not have to override, just add to; and those that do override won't have to override extensively.

**Comment by jeddy3** [comment]
> the goal should be to create a config that most users will not have to override, just add to; and those that do override won't have to override extensively

That's a far more succinct—and far better—way of putting it :)
